### ISSUE TITLE
Apache Persistence

### ISSUE THREAD

**Issue by h00die** [rapid7/metasploit-framework#17728]
https://github.com/wireghoul/htshells

I gave this a go a while ago and couldn't quite get it working, however I always thought this would be a neat module since it would allow for hiding in plain sight.  I also remember @egypt thinking something similar to this would be a good idea back at the Austin hackathon.  I think they wanted something more like a compiled apache module, but this is a start in the right direction.

**Comment by SaiSakthidar** [comment]
this seems interesting
Ill give this a shot 
Should I be trying for all shell types from htshells tho?
@h00die

**Comment by h00die** [comment]
The general idea (credits to @egypt for putting this in my brain during the AHA meeting a bunch of years back) would be to write/append to an `.htaccess` file on the system we already have access to.

## Trigger
The module to deploy it, ideally, would allow you to configure the trigger, examples:
1. `GET` to `/give_me_a_shell`
2. `POST` `/login` `username=give_me_a_shell&password=right_now`.

So I think you'd need a datastore option for `verb`, `url`, and `data`. If needed, just start with a `GET` request as that's much easier. Then someone can come back and expand it if needed.

Lastly, we'd want to either embed the payload in the .htaccess file, or have it launch an external payload. That's the typical avenue we try to have for persistence mechanisms. I think its safe to make this just a linux module for the time being.

**Comment by 4ravind-b** [comment]
Hi @h00die, saw you mention this in #gsoc. I'd like to work on this. To confirm — should I start with just GET trigger support for now, with the payload embedded in .htaccess? Happy to iterate from there.

**Comment by h00die** [comment]
yup, that would the best best idea. keep it simple to start with

**Comment by 4ravind-b** [comment]
Hi @h00die, I opened an initial implementation in #21473 following the simpler GET-trigger approach you suggested. Would appreciate any feedback or suggestions for improving/extending it further.
